File: leetcode/Python/kth-smallest-element-in-a-bst.py
# Definition for a binary tree node.
# class TreeNode:
#     def __init__(self, val=0, left=None, right=None):
#         self.val = val
#         self.left = left
#         self.right = right
class Solution:
    def kthSmallest(self, root: Optional[TreeNode], k: int) -> int:
        v = list()
        def traverse(node):
            if node is None:
                return
            traverse(node.left)
            v.append(node.val)
            traverse(node.right)
        traverse(root)
        return v[k-1]

File: leetcode/C++/kth-smallest-element-in-a-bst.cpp
/**
 * Definition for a binary tree node.
 * struct TreeNode {
 *     int val;
 *     TreeNode *left;
 *     TreeNode *right;
 *     TreeNode() : val(0), left(nullptr), right(nullptr) {}
 *     TreeNode(int x) : val(x), left(nullptr), right(nullptr) {}
 *     TreeNode(int x, TreeNode *left, TreeNode *right) : val(x), left(left), right(right) {}
 * };
 */
class Solution {
public:
    vector<int> v;
    void traverse(TreeNode *node){
        if(node==nullptr) return;
        traverse(node->left);
        v.push_back(node->val);
        traverse(node->right);
    }
    int kthSmallest(TreeNode* root, int k) {
        traverse(root);
        return v[k-1];
    }
};